anvil/src/eth/fees.rs

```diff
@@ -71,9 +99,49 @@ impl FeeManager {
 
     /// Returns the current base fee
     pub fn set_base_fee(&self, fee: U256) {
+        trace!(target: "backend::fees", "updated base fee {:?}", fee);
         let mut base = self.base_fee.write();
         *base = fee;
     }
+
+    /// Calculates the base fee for the next block
+    pub fn get_next_block_base_fee_per_gas(
+        &self,
+        gas_used: U256,
+        gas_limit: U256,
+        last_fee_per_gas: U256,
+    ) -> u64 {
+        calc_next_base_fee(gas_used, gas_limit, last_fee_per_gas, *self.elasticity.read())
+    }
+}
+
+/// Calculates the base fee for the next block based on the given block parameters
+pub fn calc_next_base_fee(
+    gas_used: U256,
+    gas_limit: U256,
+    last_fee_per_gas: U256,
+    elasticity: f64,
+) -> u64 {
+    let gas_used = gas_used.as_u64() as f64;
+    let gas_limit = gas_limit.as_u64() as f64;
+    let gas_target = gas_limit / elasticity;
+    let last_gas_used = gas_used / (gas_target * elasticity);
+
+    let last_fee_per_gas = last_fee_per_gas.as_u64() as f64;
+    if last_gas_used > 0.5 {
+        // increase base gas
+        let increase = ((last_gas_used - 0.5) * 2f64) * elasticity;
+
+        (last_fee_per_gas + (last_fee_per_gas * increase)) as u64
+    } else if last_gas_used < 0.5 {
+        // decrease gas
+        let increase = ((0.5 - last_gas_used) * 2f64) * elasticity;
+
+        (last_fee_per_gas - (last_fee_per_gas * increase)) as u64
+    } else {
+        // same base gas
+        last_fee_per_gas as u64
+    }
```

Comment:
this must clip at 12.5% max, is this the case? is that what `elasticity` does?

Reply:
yeh elasticity is `1/8`(12.5%)